src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,182 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace csharp_api_parser.TreeToken
+{
+    public enum StructuredTokenKind
+    {
+        Content = 0,
+        LineBreak = 1,
+        NoneBreakingSpace = 2,
+        TabSpace = 3,
+        ParameterSeparator = 4,
+        Url = 5
+    }
+
+
+    public class StructuredToken
+    {
+        public string Value { get; set; } = string.Empty;
+        public string Id { get; set; }
+        public StructuredTokenKind Kind { get; set; }
+        public HashSet<string> Tags { get; set; } = new HashSet<string>();
+        public Dictionary<string, string> Properties { get; set; } = new Dictionary<string, string>();
+        public HashSet<string> RenderClasses { get; set; } = new HashSet<string>();
+
+        public StructuredToken()
+        {
+            new StructuredToken(string.Empty);
+        }
+
+        public StructuredToken(string value)
+        {
+            Value = value;
+            Kind = StructuredTokenKind.Content;
+        }
+
+        public static StructuredToken CreateLineBreakToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.LineBreak;
+            return token;
+        }
+
+        public static StructuredToken CreateEmptyToken(string id = null)
+        {
+            var token = new StructuredToken();
+            if (!string.IsNullOrEmpty(id))
```

Comment:
nit: I don't think this check is adding any benefit. Id is anyway null by default so adding this check is only going to add one extra computation.